activerecord/lib/active_record/validations/uniqueness.rb

```diff
@@ -22,6 +22,7 @@ def validate_each(record, attribute, value)
 
         relation = build_relation(finder_class, attribute, value)
         if record.persisted?
+          return if record.respond_to?("#{attribute}_changed?") && !record.send("#{attribute}_changed?")
```

Comment:
```ruby
return unless record.attribute_changed?(attribute)
```

Reply:
I would find this easier to understand as a guard clause at the start of the method, even if it means duplicating the `persisted?` check.

Reply:
We can't skip the validation if any of the attributes specified in the `:scope` option have changed.

Reply:
Thanks for your input. As I mentioned this is just PoC, but your comments are helpful.